connector/src/main/java/org/jboss/as/connector/deployers/ds/processors/DriverProcessor.java

```diff
@@ -76,7 +76,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                                 Integer.valueOf(minorVersion));
                     }
                     String driverName = deploymentUnit.getName();
-                    if (! deploymentUnit.getName().endsWith(".jar") || driverNames.size() != 1) {
+                    if ((deploymentUnit.getName().contains(".") && ! deploymentUnit.getName().endsWith(".jar")) || driverNames.size() != 1) {
```

Comment:
I can't think of any other way either. It's odd, but I think it's ok so long as it's documented and will be supported.

We use similar naming convention stuff in lots of other places. E.g. ending in ".war" triggers all sort of war-related processing, ".ear" triggers ear-related processing.